core/field_dropdown.ts

```diff
@@ -114,30 +111,29 @@ export class FieldDropdown extends Field {
    * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
    */
   constructor(
-      menuGenerator: AnyDuringMigration[][]|Function|Sentinel,
-      opt_validator?: Function, opt_config?: FieldConfig) {
+      menuGenerator: MenuGenerator, opt_validator?: Function,
+      opt_config?: FieldConfig);
+  constructor(menuGenerator: Sentinel);
+  constructor(
+      menuGenerator: MenuGenerator|Sentinel, opt_validator?: Function,
+      opt_config?: FieldConfig) {
     super(Field.SKIP_SETUP);
 
-    // If we pass SKIP_SETUP, don't do *anything* with the menu generator.
-    if (menuGenerator === Field.SKIP_SETUP) {
+    if (isMenuGenerator(menuGenerator)) {
```

Comment:
As noted in the comment for `isMenuGenerator`, there's a quirk with type guarding for `Sentinel` so I shifted the order around.

Reply:
Done!